### PR TITLE
Add user configurable pruning for drop-logs

### PR DESCRIPTION
**Pull request by shastrinator**
Accept 5-tuple filters in droplog config file.

TODO: Add more tests and test on setup

Signed-off-by: Kiran Shastri <shastrinator@gmail.com>

### CODE REVIEW

File: agent-ovs/ovs/include/PacketLogHandler.h [hunk 1]
```diff
@@ -161,6 +161,8 @@ class PacketFilterSpec: public PacketTuple {
         fields.insert(std::make_pair(field_count++,
                 std::make_pair("DestinationIPPrefixLength", "")));
     }
+    enum {
```
Comment (tomflynn) [inline agent-ovs/ovs/include/PacketLogHandler.h:164]: what is this for?

**Review by tomflynn — COMMENTED** [review]
Just some minor comments

File: agent-ovs/lib/ExtraConfigManager.cpp [hunk 2]
```diff
@@ -167,4 +176,80 @@ void ExtraConfigManager::packetDropFlowConfigUpdated(PacketDropFlowConfig &dropF
...
+            .addElement("ObserverDropPruneConfig").addElement(dropPrune->filterName).build();
+        DropPruneConfig::remove(framework, dropPruneURI);
+        mutator.commit();
+        dropPruneMap.erase(dropPrune->filterName);
+        notifyPacketDropPruneConfigListeners(dropPrune->filterName);
+        return;
+    }
+    shared_ptr<DropPruneConfig> dropPruneCfg =
+        polUni.get()->addObserverDropPruneConfig(dropPrune->filterName);
+    dropPruneCfg->unsetSrcAddress();
```
[the remark this answers is not in the document]
Comment (shastrinator) [reply agent-ovs/lib/ExtraConfigManager.cpp:196]: I agree its painful too. Add call will fetch existing object if pre-existing and in that case, I need to clear everything. 
There is currently no API to do this, we would have to add one to Genie. I can take that up as a separate commit.

File: agent-ovs/ovs/include/PacketLogHandler.h (hunk 1 above)
Comment (shastrinator) [reply agent-ovs/ovs/include/PacketLogHandler.h:164]: Missed this while cleaning up.. thanks.

File: agent-ovs/lib/FSPacketDropLogConfigSource.cpp [hunk 3]
```diff
@@ -130,7 +193,56 @@ void FSPacketDropLogConfigSource::updated(const fs::path& filePath) {
             }
             dropCfg.filePath = pathStr;
             manager->packetDropLogConfigUpdated(dropCfg);
-
+            std::shared_ptr<drop_prune_set_t> newPruneSet(new drop_prune_set_t);
+            /*Notify updated*/
+            if(properties.get_child_optional(DROP_LOG_PRUNING)) {
+                int idx=0;
+                BOOST_FOREACH(const ptree::value_type& child,
```
[the remark this answers is not in the document]
Comment (shastrinator) [reply agent-ovs/lib/FSPacketDropLogConfigSource.cpp:200]: No, will fix it.

File: agent-ovs/lib/FSPacketDropLogConfigSource.cpp [hunk 4]
```diff
@@ -101,6 +101,59 @@ static inline bool validateMacAddress(const boost::optional<string>& macAddress,
...
+        return -1;
+    }
+    if(!mask_string.empty()) {
+        return !validateMacAddress(mask_string,_targetMacMask,inputField); 
+    } else {
+        _targetMacMask = none;
+    }
+    return 0;
+}
+static int validatePrefix(int filter_idx, const std::string & arg,
```
[the remark this answers is not in the document]
Comment (shastrinator) [reply agent-ovs/lib/FSPacketDropLogConfigSource.cpp:125]: Positive cases are covered in the test I already have. Added a couple of negative checks.

**Review by tomflynn — APPROVED** [review]

**Pull request merged**